### ISSUE TITLE
2023 computer choice

### ISSUE THREAD

**Issue by 130s** [kinu-garage/hut_10sqft#647]
Likely get one.

Requirement (subject to change):
- Trackpoint is a MUST.
- (Option) SD card slot :D  -> 
- GPU should probably be Nvidia to avoid unnecessary hassle with Linux and applications.
- (Option) Thunderbolt I/F for potential of GPU enclosure https://github.com/kinu-garage/hut_10sqft/issues/647#issuecomment-1452136009
- 15 inches or larger but fit into a travel backpack w/o hassle.
- Docking station availability is a must.
- Linux bundle by the vendor?
- CPU can be challenged: Doesn't need to be Intel?

**Comment by 130s** [comment]
Asking employer if replacing laptops is an option as current one must have depreciated after 5 years. If I get a new one, this may push back.

**Comment by 130s** [comment]
Just as data point, Spec of p50 [$1801](https://drive.google.com/file/d/1yrhIevmXt1UzY1ncGe_KctxiPUnjhY5z/view?usp=share_link) I received in 2017 from a vendor that bundles Linux on to Thinkpad (LAC Portlant. Seems they went out of business...? Their website 404):
- 2 x 8GB DDR
- 1 x 480GB SSD

Got another quote with 960GB SSD. Total [$1943](https://drive.google.com/file/d/1QRV121Qvv1hlSaES8CL9suWhMjXq7Dn_/view?usp=share_link).

**Comment by 130s** [comment]
# Req
- Trackpoint is a MUST.
- Docking station availability is a must.
- Linux bundle by the vendor?
- GPU should probablly be Nvidia to avoid unnecessary hassle with Linux and applications.
- CPU can be challenged: Doesn't need to be Intel?

**Comment by 130s** [comment]
Looking into Lenovo.com to get ideas.

[Travel pack I bought in 2022/01 (NOMATIC Travel Pack- Black Water Resistant Anti-Theft 30L, amazon.com)](https://www.amazon.com/NOMATIC-Resistant-Anti-Theft-Approved-Computer/dp/B07782GG5T/ref=sr_1_3?crid=2JBVHYD8OJ4GT&keywords=NOMATIC+Travel+Pack-+Black+Water+Resistant+Anti-Theft+30L+Flight+Approved+Carry+on+Laptop+Bag+Computer+Backpack&qid=1676115779&sprefix=nomatic+travel+pack-+black+water+resistant+anti-theft+30l+flight+approved+carry+on+laptop+bag+computer+backpack%2Caps%2C128&sr=8-3&ufe=app_do%3Aamzn1.fos.c3015c4a-46bb-44b9-81a4-dc28e6d374b3) doe seem to fit 15-inch laptop but [does NOT easily fit 15.6 inch according to FAQ](https://www.amazon.com/ask/questions/Tx31MCP3M7WO5T0/ref=ask_dp_dpmw_al_hza), although I've been managing to put Thinkpad p50, 15.6 inch, in the pocket that's next to the human's back. 

Highest end gaming on lenovo.com: Legion 7i Gen 7 Intel (16") with RTX 3080 Ti $3,200:
   - Processor : 12th Generation Intel® Core™ i9-12900HX Processor (E-cores up to 3.60 GHz P-cores up to 5.00 GHz)
   - Graphic Card : NVIDIA® GeForce® RTX™ 3080 Ti Laptop GPU 16GB GDDR6
   - Memory : 32 GB DDR5-4800MHz (SODIMM) - (2 x 16 GB)
   - Storage : 2 TB SSD M.2 2280 PCIe Gen4 TLC
   - Display : 16" WQXGA (2560 x 1600), IPS, Anti-Glare, Non-Touch, HDR400, 100%sRGB, 500 nits, 165Hz, LED Backlight, Narrow Bezel, Low Blue Light

**Comment by 130s** [comment]
Can't believe just yet but only laptop on lenovo.com that they can ship with Linux might be P16s, which comes with 16 inch monitor https://www.lenovo.com/us/en/p/laptops/thinkpad/thinkpadp/thinkpad-p16s-(16-inch-intel)/len101t0012

Question is if it fits in my backpack...15.6 P50 fit, but how about this? How thick is it?

- p50: 5.8-pound, 1.16-inch-thick https://www.laptopmag.com/reviews/laptops/lenovo-thinkpad-p50#:~:text=The%205.8%2Dpound%2C%201.16%2D,%2C%207.6%2Dpound%20ThinkPad%20P70.
- p16s: 1.7kgs / 3.8lbs and 20.5mm / 0.8" thin https://www.lenovo.com/us/en/p/laptops/thinkpad/thinkpadp/thinkpad-p16s-(16-inch-intel)/len101t0012 ?? How can it come that lighter and thinner?

**Comment by 130s** [comment]
[Someone on quora](https://qr.ae/prglhZ): Try Google Colab and Kaggle unless you got RTX 3080 or higher.

> Unless you buy a laptop with a RTX 3080 GPU with 16GB VRAM, it won't be good enough for anything past hello world projects(eg. MNIST etc) in deep learning. And that will cost you around twice as much as a MacBook Pro. You can make do with a 8GB VRAM GPU but even that will be really costly and you will still be fairly bottlenecked. Also, using a laptop to train models for hours at a stretch will heat up too much and decrease the life span.
> 
> My suggestion would be to use free cloud services like Google Colab and Kaggle, that would be more than enough for learning and most of the projects. You generally get a T4/ P100 GPU with 16GB VRAM, and a decent processor and RAM.
> 
> Go for a mid range gaming laptop with config like i5(12th gen+)/ ryzen7 5800H+, 16GB RAM and a decent GPU like RTX 3050 (OPTIONAL, just for testing stuff locally, and maybe some gaming if you're interested 😅). You can get this config for somewhere around 70K INR. (50K without the GPU)
> 
> Check out the new Acer Nitro/ Asus TUF series with the above config.
> 
> Edit: with the boom of LLMs (Large Language Models) even 16GB VRAM won't be enough, sometimes not even for fine-tuning. You'd have to use tricks like small batch size (even 1!) with gradient accumulation just to be able to fit in memory.

[Someone on medium.com](https://medium.com/swlh/i-bought-a-laptop-for-deep-learning-and-now-i-mainly-use-the-cloud-d297ef4d68e0#id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6ImI0OWM1MDYyZDg5MGY1Y2U0NDllODkwYzg4ZThkZDk4YzRmZWUwYWIiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJuYmYiOjE2NzY0MDMyMTIsImF1ZCI6IjIxNjI5NjAzNTgzNC1rMWs2cWUwNjBzMnRwMmEyamFtNGxqZGNtczAwc3R0Zy5hcHBzLmdvb2dsZXVzZXJjb250ZW50LmNvbSIsInN1YiI6IjEwNDY4MDY0Mzc2Mzk4NjEyNjI3NSIsImVtYWlsIjoiZ20xMzBzQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhenAiOiIyMTYyOTYwMzU4MzQtazFrNnFlMDYwczJ0cDJhMmphbTRsamRjbXMwMHN0dGcuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJuYW1lIjoiSXNhYWMgU2FpdG8iLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tL2EvQUVkRlRwNFplcC0weTRtVEgzcmE4aFhkcWhIdnhnUHBzQXVnaDc5RUNOTmdXbkk9czk2LWMiLCJnaXZlbl9uYW1lIjoiSXNhYWMiLCJmYW1pbHlfbmFtZSI6IlNhaXRvIiwiaWF0IjoxNjc2NDAzNTEyLCJleHAiOjE2NzY0MDcxMTIsImp0aSI6IjExZTUxMjA4YzBhODQwNjkxYWQzZWZlMjNkZGNhYmYyMzYwNzM5OGEifQ.K4wPReEiR3DTHxSu2B90Tbv9HzAFazqA5rS5Pw6Tgn3XKCkiXivb8tqeaw9mHuBwtNqgcgdyXEE0BIlGNVw2tjZVxpwOjtrfloPKCBL4-OfKq5UZLqTq1D0XgHzD8RAR8fkvcfiCWxeDy2IBwVP5fBcva_JJ4bAXyLmjbKbSzUi3YwuVRUfyTMN63xf2tUkJg9EYC0mI6mMLNsOuRtSKMiQvVv50A06dktAn44CwRDIt8z8XArm4QTsfr4A2-tOhRfeEnZugWGswOmlJbPa3fW8jO8A2L03XAmyQ75wFhPY1KolSHLWh573WlnGPcFomRgJESaOyeqAxo5rgzil83A) is regretting buying a laptop (or it could even be for desktop) for machine learning. Consider cloud services.

Done for "ML: laptop or cloud?" question. For my usecase cloud with seasonal $10/month or so expense is fine with so much advantage.

**Comment by 130s** [comment]
[Someone on medium.com](https://medium.com/swlh/i-bought-a-laptop-for-deep-learning-and-now-i-mainly-use-the-cloud-d297ef4d68e0#id_token=eyJhbGciOiJSUzI1NiIsImtpZCI6ImI0OWM1MDYyZDg5MGY1Y2U0NDllODkwYzg4ZThkZDk4YzRmZWUwYWIiLCJ0eXAiOiJKV1QifQ.eyJpc3MiOiJodHRwczovL2FjY291bnRzLmdvb2dsZS5jb20iLCJuYmYiOjE2NzY0MDMyMTIsImF1ZCI6IjIxNjI5NjAzNTgzNC1rMWs2cWUwNjBzMnRwMmEyamFtNGxqZGNtczAwc3R0Zy5hcHBzLmdvb2dsZXVzZXJjb250ZW50LmNvbSIsInN1YiI6IjEwNDY4MDY0Mzc2Mzk4NjEyNjI3NSIsImVtYWlsIjoiZ20xMzBzQGdtYWlsLmNvbSIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhenAiOiIyMTYyOTYwMzU4MzQtazFrNnFlMDYwczJ0cDJhMmphbTRsamRjbXMwMHN0dGcuYXBwcy5nb29nbGV1c2VyY29udGVudC5jb20iLCJuYW1lIjoiSXNhYWMgU2FpdG8iLCJwaWN0dXJlIjoiaHR0cHM6Ly9saDMuZ29vZ2xldXNlcmNvbnRlbnQuY29tL2EvQUVkRlRwNFplcC0weTRtVEgzcmE4aFhkcWhIdnhnUHBzQXVnaDc5RUNOTmdXbkk9czk2LWMiLCJnaXZlbl9uYW1lIjoiSXNhYWMiLCJmYW1pbHlfbmFtZSI6IlNhaXRvIiwiaWF0IjoxNjc2NDAzNTEyLCJleHAiOjE2NzY0MDcxMTIsImp0aSI6IjExZTUxMjA4YzBhODQwNjkxYWQzZWZlMjNkZGNhYmYyMzYwNzM5OGEifQ.K4wPReEiR3DTHxSu2B90Tbv9HzAFazqA5rS5Pw6Tgn3XKCkiXivb8tqeaw9mHuBwtNqgcgdyXEE0BIlGNVw2tjZVxpwOjtrfloPKCBL4-OfKq5UZLqTq1D0XgHzD8RAR8fkvcfiCWxeDy2IBwVP5fBcva_JJ4bAXyLmjbKbSzUi3YwuVRUfyTMN63xf2tUkJg9EYC0mI6mMLNsOuRtSKMiQvVv50A06dktAn44CwRDIt8z8XArm4QTsfr4A2-tOhRfeEnZugWGswOmlJbPa3fW8jO8A2L03XAmyQ75wFhPY1KolSHLWh573WlnGPcFomRgJESaOyeqAxo5rgzil83A) is regretting buying a laptop (or it could even be for desktop) for machine learning. Consider cloud services.

Done for "ML: laptop or cloud?" question.

**Comment by 130s** [comment]
From a dialogue with someone who does GPU-intensive (AI training), who thinks going to non-GPU laptop w/long battery, and GPU desktop.

> I'm still interested in 3D simulation, so I think I need GPU. Do I run sim outside of my office? Probably. So laptop preferred so far. Then, can I upgrade GPU on laptop? Far harder than desktop. Do I need to upgrade GPU that often? I don't know anyways.
> 
> So I'm leaning toward laptop + GPU, with a risk of GPU getting outdated quickly. That's kind of why I got interested in GPU + simulation tools on cloud, like AWS. I just can't tell which direction is more economical in 3 to 5 years. 
> 
> But I probably won't seek a career path in super GPU-intensive like yours anyways, so 5-yr old GPU won't matter much? 

## Upgrading GPU on laptop: Is it really an option?

B/c I've never looked into this.

[ How to Upgrade a Laptop Graphics Card  (lifewire.com)](https://www.lifewire.com/upgrade-graphics-card-in-laptop-5118159) reads:
> - The easiest way to upgrade your laptop's graphics is to buy a new laptop.
> - Only a very few select laptops let you upgrade the graphics chip directly.
> - External GPU enclosures let you use an additional external GPU.

External enclosure is interesting. My takeaway:
- o More types of GPUs can be an option.
- x Power supply needed for the enclosure?? That would turn a laptop to a desktop, wouldn't it, unless you're comfortably carrying around the extra power supply?

That said, I'll add Thunderbolt I/F as an option to a req. Only as an option. I'll sleep on this enclosure idea.

**Comment by 130s** [comment]
## What is Monitor "nits"?
https://ultravisionledsolutions.com/blogs/news/what-are-nits-1
- Candela: Intensity of light.
- Nit: Brightness? Hmm.

Haven't read it at all but I'm indifferent for now of brightness

## Choice of GPU
Intel or Nvidia? As mentioned in the req https://github.com/kinu-garage/hut_10sqft/issues/647#issue-1110445119, in my robotics/physics intensive usecases, I should use Nvidia as some software might specifically require Nvidia. I have historically bad experience with Nvidia + Ubuntu combination, but the latest https://github.com/kinu-garage/hut_10sqft/issues/781 wasn't bad at all.

https://www.quora.com/What-is-the-difference-between-Intel-and-NVIDIA-graphics explains Nvidia for more powerfull, dedicated usecases while Intel iGPU may be suitable for less demanding apps.

## Processor
Lenovo has 12th gen i5, i7, with variations within each. Turned out the choise of GPU depends on the choice of CPU, and Nvidia GPU requires certain types of CPU.

As dumb as can be, just following https://www.youtube.com/watch?v=GLSPub4ydiM and chose the highest number i7.

## SD card reader

["Smart Card Reader" seems to be NOT SD card reader (lenovo.com)](https://forums.lenovo.com/t5/ThinkPad-T400-T500-and-newer-T/Purchasing-I-confused-an-SD-Card-Reader-with-Smart-Card-reader/m-p/4396847).

SD reader is needed for various stuff but if the smart reader doesn't read SD, I'll do something.

**Comment by 130s** [comment]
[Went with P16s 16-inch with Ubuntu](https://groups.google.com/g/g130s/c/bxZkqOCDbUs).
- New experiment incl.
   - Duex Plus, external mobile monitor
   - C2G USB-C docking station

**Comment by 130s** [comment]
4 months after starting to use p16s, I'm hitting a bottleneck based on the h/w spec. https://github.com/kinu-garage/hut_10sqft/issues/862#issuecomment-1631509730 8GB of VRAM is short for what I'm trying now. So upgrading GPU on laptop https://github.com/kinu-garage/hut_10sqft/issues/647#issuecomment-1452136009 is a serious consideration.

**Comment by 130s** [comment]
[Spec sheet](https://drive.google.com/file/d/1tHcN_T79nIIp1osNTRHGEePdU5Ke6tzj/view?usp=drive_link).

**Comment by 130s** [comment]
Btw RAM upgrade is also impossible with p16s as it's soldered, confirmed on the [official doc (lenovo.com)](https://psref.lenovo.com/syspool/Sys/PDF/ThinkPad/ThinkPad_P16s_Gen_2_AMD/ThinkPad_P16s_Gen_2_AMD_Spec.pdf) via [reddit](https://www.reddit.com/r/thinkpad/comments/1frp08y/is_ram_upgradeable_in_thinkpad_p16s/).
